src/query/procedure/module.cpp

```diff
@@ -320,6 +322,350 @@ void RegisterMgTransformations(const std::map<std::string, std::unique_ptr<Modul
   module->AddProcedure("transformations", std::move(procedures));
 }
 
+namespace {
+bool IsAllowedExtension(const auto &extension) {
+  constexpr std::array<std::string_view, 1> allowed_extensions{".py"};
+  return std::any_of(allowed_extensions.begin(), allowed_extensions.end(),
+                     [&](const auto allowed_extension) { return allowed_extension == extension; });
+}
+
+bool IsSubPath(const auto &base, const auto &destination) {
+  const std::string relative = std::filesystem::relative(destination, base);
+  // if it starts with '..', base is not subpath of destination
+  return relative.size() == 1 || relative[0] != '.' || relative[1] != '.';
+}
+
```

Comment:
This is such an annoying problem that seems easy at first...
You are correct, I need to check if it's empty, some examples in the docs show how it can happen (even though I'm not sure how to trigger it on Linux - `/etc` relative to `/other` will return `../other`)

Your `IsSubPath2` looks okay, the current solution tries to shortcircuit as much as possible but there is no reason to do be so pedantic here.